src/server/coprocessor/endpoint.rs

```diff
@@ -105,20 +105,60 @@ impl Display for Task {
 
 pub struct RequestTask {
     req: Request,
+    start_ts: Option<u64>,
+    wait_time: Option<f64>,
+    timer: Instant,
     // The deadline before which the task should be responded.
     deadline: Instant,
     on_resp: OnResponse,
 }
 
 impl RequestTask {
     pub fn new(req: Request, on_resp: OnResponse) -> RequestTask {
-        let deadline = Instant::now() + Duration::from_secs(REQUEST_MAX_HANDLE_SECS);
+        let timer = Instant::now();
+        let deadline = timer + Duration::from_secs(REQUEST_MAX_HANDLE_SECS);
         RequestTask {
             req: req,
+            start_ts: None,
+            wait_time: None,
+            timer: timer,
             deadline: deadline,
             on_resp: on_resp,
         }
     }
+
+    #[inline]
+    fn check_outdated(&self) -> Result<()> {
+        check_if_outdated(self.deadline, self.req.get_tp())
+    }
+
+    fn start_handling(&mut self) {
+        if self.wait_time.is_some() {
+            return;
+        }
+        let wait_time = duration_to_sec(self.timer.elapsed());
+        COPR_REQ_WAIT_TIME.with_label_values(&["select", get_req_type_str(self.req.get_tp())])
+            .observe(wait_time);
+        self.wait_time = Some(wait_time);
+    }
+
+    fn finish_handling(&mut self) {
+        self.start_handling();
```

Comment:
In case `start_handling` is not called yet. Calling it here to update wait time.